File: backend/app/schemas/supplier.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class SupplierBase(BaseModel):
    name: str
    phone: Optional[str] = None
    address: Optional[str] = None
    notes: Optional[str] = None


class SupplierCreate(SupplierBase):
    pass


class SupplierUpdate(SupplierBase):
    name: Optional[str] = None


class SupplierResponse(SupplierBase):
    id: int
    business_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/lib/api/supplier.ts
import { apiClient } from './client'

export interface Supplier {
  id: number
  business_id: number
  name: string
  phone?: string
  address?: string
  notes?: string
  created_at: string
  updated_at: string
}

export interface SupplierCreate {
  name: string
  phone?: string
  address?: string
  notes?: string
}

export const supplierApi = {
  async getSuppliers(): Promise<Supplier[]> {
    const response = await apiClient.get('/suppliers')
    return response.data
  },

  async getSupplier(id: number): Promise<Supplier> {
    const response = await apiClient.get(`/suppliers/${id}`)
    return response.data
  },

  async createSupplier(data: SupplierCreate): Promise<Supplier> {
    const response = await apiClient.post('/suppliers', data)
    return response.data
  },

  async updateSupplier(id: number, data: Partial<SupplierCreate>): Promise<Supplier> {
    const response = await apiClient.put(`/suppliers/${id}`, data)
    return response.data
  },

  async deleteSupplier(id: number): Promise<void> {
    await apiClient.delete(`/suppliers/${id}`)
  },
}


